### ISSUE TITLE
1.8 ansible-galaxy overly strict requirements.txt parsing

### ISSUE THREAD

**Issue by ches** [ansible/ansible#9743]
##### Issue Type:

Bug Report / Feature Request

Hash comments used to work, not sure if breaking them is considered a bug if they weren't _documented_ as supported previously, so if not, consider it a request—comments should be supported :stuck_out_tongue_closed_eyes:
##### Ansible Version:

1.8.2
##### Environment:

Control host is OS X, managed is N/A.
##### Summary:

A `requirements.txt` containing blank lines and some lines beginning with `#` as comments worked as expected in 1.7.x but throws exceptions in 1.8. Even a single blank line at end of the file causes an exception, though if that's the only one then it's spurious because all of the work is done.
##### Steps To Reproduce:

Create `requirements.txt` with the following contents:

```
# We're on an old version because of blah blah...
Stouts.jenkins,1.2.0
```

Include a blank line with newline at the end too.

```
$ ansible-galaxy install -r requirements.txt
```
##### Expected Results:

Successful role installation without errors.
##### Actual Results:

```
$ ansible-galaxy install -r roles/requirements.txt
Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/1.8.2/libexec/bin/ansible-galaxy", line 954, in <module>
    main()
  File "/usr/local/Cellar/ansible/1.8.2/libexec/bin/ansible-galaxy", line 948, in main
    fn(args, options, parser)
  File "/usr/local/Cellar/ansible/1.8.2/libexec/bin/ansible-galaxy", line 778, in execute_install
    role_src = role.get("src")
AttributeError: 'tuple' object has no attribute 'get'
```

Remove the comment but leave the trailing newline, and the role will install but you'll still get the ugly exception spewed to output.

**Comment by mpdehaan** [comment]
I don't think this is a regression because there wasn't an old bug that was fixed and this was re-broken, probably, but I agree this should probably work.

We'll take a look -- thanks!

**Comment by lukeorland** [comment]
Another example of overly strict parsing  is when the requirements file has no uncommented lines to parse. E.g.:

```
$ cat requirements.yml
---
#- src: goozbach.EPEL

$ ansible-galaxy install -r requirements.yml
Traceback (most recent call last):
  File "/usr/local/bin/ansible-galaxy", line 959, in <module>
    main()
  File "/usr/local/bin/ansible-galaxy", line 953, in main
    fn(args, options, parser)
  File "/usr/local/bin/ansible-galaxy", line 767, in execute_install
    roles_left = map(ansible.utils.role_yaml_parse, yaml.safe_load(f))
TypeError: argument 2 to map() must support iteration
```

Ansible 1.9.2

**Comment by bcoca** [comment]
fixed in devel via 0e778714262479258bd612910469e20e674e8a3b
